desktop-widgets/command_divelist.cpp

```diff
@@ -918,8 +918,13 @@ MergeDives::MergeDives(const QVector <dive *> &dives)
 	d->divetrip = nullptr;
 	d->dive_site = nullptr;
 
-	// The merged dive gets the number of the first dive
-	d->number = dives[0]->number;
+	// The merged dive gets the number of the first dive with a non-zero number
+	for (const dive *dive: dives) {
+		if (dive->number) {
+			d->number = d->number;
```

Comment:
that does seem a bit too tautological... I think you meant `d->number = dive->number`...